learn/sherif/main_test/src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -1,40 +1,48 @@
 package frc.robot.subsystems;
 
-import edu.wpi.first.wpilibj.SpeedController;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-import java.math.BigDecimal;
-import java.math.RoundingMode;
-
 import static frc.robot.Constants.*;
 
 public class DriveTrain extends SubsystemBase {
-    private SpeedController left;
-    private SpeedController right;
+    private final WPI_TalonSRX m_left_a;
+    private final WPI_TalonSRX m_left_b;
+    private final WPI_TalonSRX m_left_c;
 
-    public DriveTrain(Factory f) {
-        this.left = f.getSparkMotor(FRONT_MOTOR);
-        this.right = f.getSparkMotor(BACK_MOTOR);
-        this.right.getInverted();
-    }
+    private final SpeedControllerGroup left;
 
-    public void drive(double ySpeed, double xSpeed) {
-        this.right.set((Clip(ySpeed + xSpeed)));
-        this.left.set((Clip(ySpeed - xSpeed)));
-    }
+    private final WPI_TalonSRX m_right_a;
+    private final WPI_TalonSRX m_right_b;
+    private final WPI_TalonSRX m_right_c;
 
-    public void brake() {
-        this.right.set(0.0);
-        this.left.set(0.0);
+    private final SpeedControllerGroup right;
+
+    private final DifferentialDrive differentialDrive;
+
+    public DriveTrain(Factory f) {
+        this.m_left_a = f.getTalonMotor(DRIVE_LEFT_MOTOR_A);
+        this.m_left_b = f.getTalonMotor(DRIVE_LEFT_MOTOR_B);
+        this.m_left_c = f.getTalonMotor(DRIVE_LEFT_MOTOR_C);
+        this.left = new SpeedControllerGroup(this.m_left_a, this.m_left_b, this.m_left_c);
+
+        this.m_right_a = f.getTalonMotor(DRIVE_RIGHT_MOTOR_A);
+        this.m_right_b = f.getTalonMotor(DRIVE_RIGHT_MOTOR_B);
+        this.m_right_c = f.getTalonMotor(DRIVE_RIGHT_MOTOR_C);
+        this.right = new SpeedControllerGroup(this.m_right_a, this.m_right_b, this.m_right_c);
+        this.right.setInverted(true);
+
+        this.differentialDrive = new DifferentialDrive(this.left, this.right);
+        this.differentialDrive.setDeadband(0.0);
     }
 
-    public static double Clip(double value, double min, double max) {
-        if (value < min) return min;
-        return Math.min(value, max);
+    public void drive(double xSpeed, double rot) {
+        this.differentialDrive.arcadeDrive(xSpeed, rot, false);
     }
 
-    public static double Clip(double value) {
-        if (value < -1.0) return -1.0;
-        return Math.min(value, 1.0);
+    public void brake() {
+        this.differentialDrive.tankDrive(0, 0, false);
```

Comment:
This isn't connected to the controllers anywhere and I don't think you need this with arcade drive or tank drive since the robot will stop when the joystick goes back to neutral